HW4_Voskoboinikov/ultimate_protein_tools.py

```diff
@@ -0,0 +1,691 @@
+    """
+
+    check_upper = True
+    for letter in (set(codon)):
+        check_upper = letter.isupper() and check_upper
+    return check_upper
+
+
+def get_frameshift_proteins(seq: int,
+                            check_if_user_conscious: bool = False,
+                            is_stop_codon_termination_enabled: bool = False,
+                            **_) -> dict:
+    """
+    Returns list of all possible proteins from all possible frames in peptide.
+
+    WARNING: can be computationally intensive on longer sequences,
+    will NOT start unless check_if_user_conscious is True!
+    
+    Arguments:
+    - seq (str): seq to be checked
+    - check_if_user_conscious (bool): checks user's consciousness. Default False
+    - is_stop_codon_termination_enabled (bool): terminate translation when reached stop-codon. Default False.
+
+    Return:
+    - dict: dict of lists of all possible frames proteins:
+    {frame_0: ['protein_seqs'], frame_1: ['protein_seqs'], frame_2: ['protein_seqs']}
+    """
+
+    if check_if_user_conscious:
+        frameshift_dct = {'frame_0': [seq]}  # set current seq as frame_0 (protein from not-shifted frame)
+        rnas = get_protein_rnas(seq, check_if_user_conscious=check_if_user_conscious)
+        for frame_number in [1, 2]:
+            frames_list = []
+            for rna in rnas:
+                frame = ''
+                for i in range(frame_number, len(rna) - (frame_number + 1), 3):  # set frame-dependent range to iterate
+                    frame_codon = rna[i:i + 3]  # extract codon
+                    if not check_all_upper(frame_codon):  # check if all letters in codon uppercase
+                        frame_codon = frame_codon.lower()  # if not change all to lowercase
+                    frame += RNA_CODON_TABLE[frame_codon]
+                    if is_stop_codon_termination_enabled and RNA_CODON_TABLE[
+                        frame_codon] == '*':  # stop writing if meet stop-codon
+                        break
+                frames_list.append(frame)  # append frame to frames list
+            frameshift_dct[f'frame_{frame_number}'] = list(set(frames_list))  # clean duplicates and write to dict
+        return frameshift_dct
+    return "You don't fucking know what you're doing!"  # politely ask user to reconsider their actions
+
+
+def get_length_of_protein(seq: str, **_) -> int:
+    """
+    Calculates the length of a protein.
+
+    Argument:
+    - seq (str): sequence to calculate the length
+
+    Return:
+    - int: sequence length
+    """
+
+    return len(seq)
+
+
+def count_aa(seq: str, aminoacids: str = None, **_) -> dict:
+    """
+    Counts the number of given or all amino acids in a protein sequence.
+
+    Arguments:
+    - seq (str): sequence to count amino acids
+    - aminoacids (str): which amino acids to count in sequence
+
+    Return:
+    - dict: a dictionary with amino acids and its count
+    """
+
+    aa_dict_count = {}
```

Comment:
1) Множественное число
2) Не надо писать тип данных в названии переменной. Мы же не пишем `for i_int in range(0,10):`

```suggestion
    aas_counts = {}
```